datasets/flwr_datasets/partitioner/iid_partitioner.py

```diff
@@ -47,11 +50,32 @@ def load_partition(self, partition_id: int) -> datasets.Dataset:
         dataset_partition : Dataset
             single dataset partition
         """
-        return self.dataset.shard(
-            num_shards=self._num_partitions, index=partition_id, contiguous=True
-        )
+        self._determine_partition_id_to_indices_if_needed()
+        return self.dataset.select(self._partition_id_to_indices[partition_id])
```

Comment:
the `.shard()` approach above had to set `contiguous` to `True`, which is not the default value. Do we need some special handling because we are now using [`.select()`](https://huggingface.co/docs/datasets/v2.20.0/en/package_reference/main_classes#datasets.Dataset.select). I guess no?

Reply:
No, the change I did matches the behavior of shard with `contiguous=True` and it's based on the implemention of `shard`. So the order of samples will remain the same after this change.